Submitted By: Jim Gifford (patches at jg555 dot com)
Date: 2006-01-12
Initial Package Version: 2.6.15
Origin: Jim Gifford
Upstream Status: Unknown
Description: Fixes Build issue on Sparc

--- linux-2.6.15/arch/sparc64/kernel/process.c.orig	2006-01-12 13:27:03.000000000 -0800
+++ linux-2.6.15/arch/sparc64/kernel/process.c	2006-01-12 13:28:21.000000000 -0800
@@ -164,6 +164,7 @@
 	panic("Reboot failed!");
 }
 
+#ifdef CONFIG_COMPAT
 static void show_regwindow32(struct pt_regs *regs)
 {
 	struct reg_window32 __user *rw;
@@ -189,6 +190,7 @@
 	       r_w.ins[0], r_w.ins[1], r_w.ins[2], r_w.ins[3],
 	       r_w.ins[4], r_w.ins[5], r_w.ins[6], r_w.ins[7]);
 }
+#endif
 
 static void show_regwindow(struct pt_regs *regs)
 {
@@ -214,7 +216,11 @@
 			set_fs (old_fs);			
 		}
 	} else {
-		show_regwindow32(regs);
+		#ifdef CONFIG_COMPAT
+			show_regwindow32(regs);
+		#else
+			show_regwindow(regs);
+		#endif
 		return;
 	}
 	printk("l0: %016lx l1: %016lx l2: %016lx l3: %016lx\n",
